libraries/AP_HAL/AP_HAL_Boards.h

```diff
@@ -325,3 +325,11 @@
 #ifndef HAL_ENABLE_DFU_BOOT
 #define HAL_ENABLE_DFU_BOOT 0
 #endif
+
+
+// sanity checks for the configuration.  This can't test everything as
+// the libraries can do their own definitions - but we can catch some
+// things:
+#if HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024
+#error "2MB board with minimize features?!"
```

Comment:
no big deal but I think we should keep the wording more formal so I think we should drop the "?!" in this message.  Some alternative wordings are:

- 2MB board should not be compiled with HAL_MINIMIZE_FEATURES
- 2MB board should be compiled with HAL_MINIMIZE_FEATURES = 0